scene/main/viewport.cpp

```diff
@@ -5218,12 +5218,15 @@ Size2i SubViewport::get_size() const {
 
 void SubViewport::set_size_2d_override(const Size2i &p_size) {
 	ERR_MAIN_THREAD_GUARD;
-	_set_size(_get_size(), p_size, true);
+	_set_size(_get_size(), Size2(p_size), true);
 }
 
 Size2i SubViewport::get_size_2d_override() const {
 	ERR_READ_THREAD_GUARD_V(Size2i());
-	return _get_size_2d_override();
+	// Rounding will cause offset issues with the
+	// exact positioning of subwindows, but changing the
+	// type of size_2d_override would break compatibility.
+	return Size2i((_get_size_2d_override() + Size2(0.5, 0.5)).floor());
```

Comment:
Suggesting reversion. The MRP is fixed without `size_2d_override` being a `Size2`.

```suggestion
	return _get_size_2d_override();
```